src/accelerate/data_loader.py

```diff
@@ -89,7 +89,9 @@ def __init__(self, *args, **kwargs):
 
     def __iter__(self):
         if self.generator is None:
-            self.generator = torch.Generator(device=torch.get_default_device())
+            self.generator = torch.Generator(
+                device=torch.get_default_device() if hasattr(torch, "get_default_device") else None
```

Comment:
Note that according to the [docs](https://pytorch.org/docs/stable/generated/torch.Generator.html), the default value of `device` for `Generator` is `"cpu"`. Maybe `None` would also work but I think `"cpu"` is the safer bet.